src/Components/Login.js

```diff
@@ -0,0 +1,203 @@
+import {
+  StyledH1,
+  StyledLink,
+  StyledPLabel
+} from './styled-components/StyledText';
+import Spinner from './semantic-components/Spinner';
+import LoginAnimation from './animations/LoginAnimation';
+import { PasswordlessButton } from './styled-components/StyledButton';
+import showPassword from '../images/showPassword.svg';
+
+class Login extends Component {
+  static propTypes = {
+    auth: PropTypes.object,
+    firebase: PropTypes.shape({
+      login: PropTypes.func.isRequired,
+      logout: PropTypes.func.isRequired
+    })
+  };
+
+  state = {
+    loginEmail: '',
+    loginPassword: '',
+    error: null
+  };
+
+  componentWillUpdate() {
+    if (!isEmpty(this.props.auth)) {
+      this.props.history.push('/homescreen');
+    }
+  }
+
+  handleInputChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  togglePassword = () => {
+    let temp = document.getElementById('typepass');
+    if (temp.type === 'password') {
+      temp.type = 'text';
+    } else {
+      temp.type = 'password';
+    }
+  };
+
+  passwordlessSignIn = loginEmail => {
+    const actionCodeSettings = {
+      url: 'http://localhost:3000/passwordlesscheck',
+      handleCodeInApp: true
+    };
+
+    const INITIAL_STATE = {
+      loginEmail: '',
+      loginPassword: '',
+      error: null
+    };
+
+    this.props.firebase
+      .auth()
+      .sendSignInLinkToEmail(loginEmail, actionCodeSettings)
+      .then(function() {
+        window.localStorage.setItem('emailForSignIn', loginEmail);
+      })
+      .then(() => {
+        this.setState({ ...INITIAL_STATE });
+      })
+      .catch(error => {
+        this.setState({ error: error.message }); // not working
+      });
+  };
+
+  render() {
+    const { loginEmail, loginPassword } = this.state;
+    const isInvalid = loginPassword === '' || loginEmail === '';
+    const passwordlessIsInvalid = loginEmail === '';
+
+    if (!isLoaded(this.props.auth)) {
+      return <Spinner />;
+    }
+    if (!isEmpty(this.props.auth)) {
+      return null;
+    }
+    return (
+      <StyledLogin>
+        <StyledLoginCon>
+          <StyledH1>Sign in</StyledH1>
+          <StyledForm>
+            <StyledLabel>
+              <StyledPLabel>Email Address</StyledPLabel>
+              <StyledInput
+                name='loginEmail'
+                type='email'
+                onChange={this.handleInputChange}
+                placeholder='tonystark@example.com'
+              />
+            </StyledLabel>
+            <StyledLabel>
+              <StyledPLabel>Password</StyledPLabel>
+              <StyledInput
+                id='typepass'
+                name='loginPassword'
+                type='password'
+                onChange={this.handleInputChange}
+                placeholder='········'
+              />
+              <StyledIcon
+                src={showPassword}
+                alt='showPassword'
+                onClick={this.togglePassword}
+              />
```

Comment:
can u change the icon if you can't see the password like a give it a grey color or change the icon